src/app/api/UserAuthentication_service/UserAuthentication.service.ts

```diff
@@ -0,0 +1,76 @@
+import { Injectable } from '@angular/core';
+
+import { User } from '../Classes/user';
+import { Weather } from '../Classes/weather';
+import { WeatherAPIService } from '../WeatherAPI_service/WeatherAPI.service';
+
+@Injectable()
+export class UserAuthenticationService {
+
+  private isUserLoggedIn;
+
+  constructor() {
+    this.isUserLoggedIn = false;
+  }
+
+  setUserLoggedIn(bool: boolean) {
+    this.isUserLoggedIn = bool;
+  }
+
+  getUserLoggedIn(): boolean {
+    return this.isUserLoggedIn;
+  }
+
+  getCurrentUser(): User {
+    return JSON.parse(localStorage.getItem('currentUser'));
+  }
+
+  userExists (email: string, password: string): boolean {
+    const users = JSON.parse(localStorage.getItem('users')) || [];
+    const filteredUsers = users.filter((u) => {
+      return u.email === email && u.password === password;
+    });
+
+    return filteredUsers.length > 0 ? true : false ;
+  }
+
+  setCurrentUser(email: string, password: string) {
+    const users = JSON.parse(localStorage.getItem('users'));
+    let user: User;
+    for (let u of users) {
```

Comment:
there is no need to use unreadable character long variables like `u` we have uglifiers these days to compress everything :)